driver/sigmoid_focal_loss_driver.hpp

```diff
@@ -0,0 +1,797 @@
+            // L = ce_loss * pow_pt => dL/dt = dceloss/dt * pow_pt + ce_loss * dpowpt/dt
+            Tcheck dLdt       = dcelossdt * powPt + ceLoss * dpowptdt;
+            Tcheck gradTarget = dO * dLdt;
+
+            if(alpha >= 0)
+            {
+                // alpha_t * dL/dt + dalpha_t/dt * dL
+                gradTarget = alpha_t * dLdt + (2 * alpha - 1) * ceLoss * powPt;
+            }
+            dtarget[dtarget_tv.get_tensor_view_idx(idx)] = static_cast<Tcheck>(gradTarget);
+        }
+    }
+}
+
+template <typename Tgpu, typename Tcheck>
+void mloSigmoidFocalLossFwdRunHost(Tgpu* input,
+                                   miopenTensorDescriptor_t inputDesc,
+                                   Tgpu* target,
+                                   miopenTensorDescriptor_t targetDesc,
+                                   Tcheck* workspaceHost,
+                                   Tcheck* outputHost,
+                                   float alpha   = 0.25,
+                                   float gamma   = 2,
+                                   float divisor = 1)
+{
+    auto input_tv    = miopen::get_inner_expanded_tv<5>(miopen::deref(inputDesc));
+    auto target_tv   = miopen::get_inner_expanded_tv<5>(miopen::deref(targetDesc));
+    size_t inputSize = miopen::deref(inputDesc).GetElementSize();
+
+    for(size_t id = 0; id < inputSize; ++id)
+    {
+        tensor_layout_t<5> idx(input_tv, id);
+
+        Tcheck i = static_cast<Tcheck>(input[input_tv.get_tensor_view_idx(idx)]);
+        Tcheck t = static_cast<Tcheck>(target[target_tv.get_tensor_view_idx(idx)]);
+
+        Tcheck sig    = 1 / (1 + exp(-i));
+        Tcheck ceLoss = -(t * log(sig) + (1 - t) * log(1 - sig));
+        Tcheck sigT   = sig * t + (1 - sig) * (1 - t);
+        Tcheck loss   = ceLoss * pow(1 - sigT, gamma);
+
+        if(alpha >= 0)
+        {
+            Tcheck alphaT = alpha * t + (1 - alpha) * (1 - t);
+            loss          = alphaT * loss;
+        }
+
+        workspaceHost[id] = static_cast<Tcheck>(loss / divisor);
+    }
+
+    // Reduce loss
+    const int local_size = 256;
+    int offset_a         = 0;
+    int offset_b         = inputSize;
+    size_t _size         = inputSize;
+    do
+    {
+        for(int i = 0; i < _size; i += local_size)
+        {
+            Tcheck shared[local_size];
+            for(int j = 0; j < local_size; ++j)
+                shared[j] = i + j < _size ? workspaceHost[offset_a + i + j] : 0.0f;
+            for(int offset = local_size / 2; offset > 0; offset >>= 1)
+                for(int j = 0; j < offset; ++j)
+                    shared[j] += shared[j + offset];
+            if(_size <= local_size)
+                outputHost[0] = shared[0];
+            else
+                workspaceHost[offset_b + i / local_size] = shared[0];
+        }
+        std::swap(offset_a, offset_b);
+        _size = (_size + local_size - 1) / local_size;
+    } while(_size > 1);
```

Comment:
```suggestion
```

Reply:
Is this a mistake? I saw no differences between 2 versions.

Reply:
Same as the cmt above.